pkg/wekafs/apistore.go

```diff
@@ -101,30 +108,42 @@ func (api *ApiStore) fromSecrets(ctx context.Context, secrets map[string]string,
 
 	kmsVaultNamespaceForFilesystemEncryption, ok := secrets["kmsVaultNamespaceForFilesystemEncryption"]
 	if ok {
-		preexistingVaultCreds.Namespace = strings.TrimSpace(strings.TrimSuffix(kmsVaultNamespaceForFilesystemEncryption, "\n"))
+		preexistingVaultCreds.Namespace = cleanSecretValue(kmsVaultNamespaceForFilesystemEncryption)
 	}
 
 	kmsVaultKeyIdentifierForFilesystemEncryption, ok := secrets["kmsVaultKeyIdentifierForFilesystemEncryption"]
 	if ok {
-		preexistingVaultCreds.KeyIdentifier = strings.TrimSpace(strings.TrimSuffix(kmsVaultKeyIdentifierForFilesystemEncryption, "\n"))
+		preexistingVaultCreds.KeyIdentifier = cleanSecretValue(kmsVaultKeyIdentifierForFilesystemEncryption)
 	}
 
 	kmsVaultRoleIdForFilesystemEncryption, ok := secrets["kmsVaultRoleIdForFilesystemEncryption"]
 	if ok {
-		preexistingVaultCreds.RoleId = strings.TrimSpace(strings.TrimSuffix(kmsVaultRoleIdForFilesystemEncryption, "\n"))
+		preexistingVaultCreds.RoleId = cleanSecretValue(kmsVaultRoleIdForFilesystemEncryption)
 	}
 
 	kmsVaultSecretIdForFilesystemEncryption, ok := secrets["kmsVaultSecretIdForFilesystemEncryption"]
 	if ok {
-		preexistingVaultCreds.SecretId = strings.TrimSpace(strings.TrimSuffix(kmsVaultSecretIdForFilesystemEncryption, "\n"))
+		preexistingVaultCreds.SecretId = cleanSecretValue(kmsVaultSecretIdForFilesystemEncryption)
+	}
+	username, ok := secrets["username"]
+	if !ok {
+		return nil, fmt.Errorf("no username found in secret")
+	}
+	password, ok := secrets["password"]
+	if !ok {
+		return nil, fmt.Errorf("no password found in secret")
+	}
+	organization, ok := secrets["organization"]
```

Comment:
@tigrawap suggested to add both "organization" and "org"
Could be usable but on the other hand, so also add "user", "pass", "pw"?

I am sceptic.

Reply:
why "org"? It's supposed to be "organization" no?
added password and user

Reply:
that's fine, drop the org(==no changes). as long as it's clear what's missing no need to lookup from multiple places